app/screens/find_channels/filtered_list/index.ts

```diff
@@ -37,11 +37,14 @@ const enhanced = withObservables(['term'], ({database, term}: EnhanceProps) => {
         switchMap((matchStart) => {
             return retrieveChannels(database, matchStart.flat(), true);
         }),
+        switchMap((myChannels) => removeChannelsFromArchivedTeams(myChannels, teamIds)),
     );
 
     const channelsMatch = joinedChannelsMatch.pipe(
         combineLatestWith(directChannelsMatch),
         switchMap((matched) => retrieveChannels(database, matched.flat(), true)),
+        switchMap((myChannels) => removeChannelsFromArchivedTeams(myChannels, teamIds)),
```

Comment:
Instead of passing the observable as an argument, you can use here `combineLatestWith`. That way this would end up something like...

```suggestion
        combineLatestWith(teamIds),
        switchMap(([myChannels, teamIds]) => of$(removeChannelsFromArchivedTeams(myChannels, teamIds))),
```